From 9f3474cf0b4355be870d980afc994f52a02ce70b Mon Sep 17 00:00:00 2001
Message-Id: <9f3474cf0b4355be870d980afc994f52a02ce70b.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Jeff McGee <jeff.mcgee@intel.com>
Date: Thu, 23 Jan 2014 15:10:02 -0600
Subject: [PATCH 063/312] MUST_REBASE [VPG]: drm/i915: Set backlight with DPST

The backlight is being set directly (without DPST adjustment factor)
by the *_enable_backlight functions. Fix this in pch_enable_backlight
so that DPST adjustments are preserved for Haswell on a disable-
enable cycle.

Issue: APDEV-1279
Change-Id: I9fd3d6f056fe121ddf9c2a4b609d98bfb6e76caa
Signed-off-by: Jeff McGee <jeff.mcgee@intel.com>
Reviewed-on: https://android.intel.com/161786
Tested-by: sys_abtbuild <sys_abtbuild@intel.com>
Reviewed-by: Noziska, Patrick J <patrick.j.noziska@intel.com>
Reviewed-by: S, Deepak <deepak.s@intel.com>
Reviewed-by: Gurram, Sudheer <sudheer.gurram@intel.com>
Reviewed-by: Azad, Vinit <vinit.azad@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/intel_panel.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_panel.c b/drivers/gpu/drm/i915/intel_panel.c
index 96d7c1b..d25c15f 100644
--- a/drivers/gpu/drm/i915/intel_panel.c
+++ b/drivers/gpu/drm/i915/intel_panel.c
@@ -745,7 +745,11 @@ static void pch_enable_backlight(struct intel_connector *connector)
 	I915_WRITE(BLC_PWM_CPU_CTL2, cpu_ctl2 | BLM_PWM_ENABLE);
 
 	/* This won't stick until the above enable. */
-	intel_panel_actually_set_backlight(connector, panel->backlight.level);
+	if (dev_priv->dpst.enabled)
+		i915_dpst_set_brightness(dev, panel->backlight.level);
+	else
+		intel_panel_actually_set_backlight(connector,
+						   panel->backlight.level);
 
 	pch_ctl2 = panel->backlight.max << 16;
 	I915_WRITE(BLC_PWM_PCH_CTL2, pch_ctl2);
-- 
1.7.9.5

